dali/pipeline/executor/executor2/exec2.cc

```diff
@@ -345,6 +351,7 @@ class Executor2::Impl {
       throw std::logic_error("Incorrect state transition.");
     exec_ = std::make_unique<tasking::Executor>(config_.operator_threads);
     exec_->Start([this](){
+      SetThreadName("[DALI] Executor");
```

Comment:
I don't know if this is the best place. Maybe you can add it to executor.h:
```
  void Start(std::function<void()> thread_setup = {}) {
    if (started_)
      return;
    assert(workers_.empty());
    for (int i = 0; i < num_threads_; i++)
      workers_.emplace_back([thread_setup, this, i]() {
        SetThreadName(make_string("[DALI] Executor", i));
        if (thread_setup)
          thread_setup();
        RunWorker();
      });
    started_ = true;
  }
```

Reply:
I wanted `tasking` to be a clean C++ library and adding DALI-specific stuff there would work against it. I specifically added `tread_setup` not to mix the usage with the `tasking::Executor` class.
Also, it's possible to have multiple instances of `tasking::Executor` running in parallel, so having an ability to name the threads by _usage_ (as opposed to naming them by _class_) seems beneficial.

Reply:
Is there any value in extending API to allow passing the thread index to the executed functions?

Reply:
To the `thread_setup` - maybe, but that's rather beyond the scope of this PR.
To the actual jobs - not really. It would just break the abstraction of the `Scheduler`.

We can consider this again when adding affinity support. Note that an attempt to fix a thread to a core may actually backfire (because the OS might try to be fair to _threads_ and then we'd jump from core to core).